src/execution_plan/optimizations/traverse_order_utils.c

```diff
@@ -272,8 +272,8 @@ void TraverseOrder_ScoreExpressions
 			score = TraverseOrder_FilterExistenceScore(exp, qg,
 													   filtered_entities);
 			if(score > 0) {
-				if(_AlgebraicExpression_IsVarLen(exp, qg)) {
-					// variable length traversal should always "lose" to its
+				if(_AlgebraicExpression_IsRangeLen(exp, qg)) {
```

Comment:
I believe this entire file should be reverted once the change to `qg_edge.h` is applied.